File: frontend/src/type/incident.ts
export type IncidentType = "fire" | "flood" | "earthquake";

export interface Incident {
    title: string;
    incident_type: IncidentType;
    description?: string | null;
    location?: string | null;
    image?: string | null;
}

export interface ListItem {
    key: string;
    incident: Incident;
}

export interface ListResponse {
    items: ListItem[];
}

File: api/src/domain/entities/incident.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
#[serde(rename_all = "snake_case")]
pub struct Incident {
    pub title: String,
    pub incident_type: IncidentType,
    pub description: Option<String>,
    pub location: Option<String>,
    pub image: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
#[serde(rename_all = "snake_case")]
pub enum IncidentType {
    Fire,
    Flood,
    Earthquake,
}
